TGM/src/main/java/network/warzone/tgm/modules/legacy/LegacyArmorModule.java

```diff
@@ -0,0 +1,155 @@
+package network.warzone.tgm.modules.legacy;
+
+import com.google.gson.JsonObject;
+import network.warzone.tgm.TGM;
+import network.warzone.tgm.match.Match;
+import network.warzone.tgm.match.MatchModule;
+import org.bukkit.Material;
+import org.bukkit.attribute.Attribute;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.inventory.EntityEquipment;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+public class LegacyArmorModule extends MatchModule implements Listener {
+
+    private static final double REDUCTION_PER_ARMOR_POINT = 0.04;
+
+    private static final Set<EntityDamageEvent.DamageCause> NON_REDUCED_CAUSES = EnumSet.of(
+            EntityDamageEvent.DamageCause.FIRE_TICK,
+            EntityDamageEvent.DamageCause.VOID,
+            EntityDamageEvent.DamageCause.SUFFOCATION,
+            EntityDamageEvent.DamageCause.DROWNING,
+            EntityDamageEvent.DamageCause.STARVATION,
+            EntityDamageEvent.DamageCause.FALL,
+            EntityDamageEvent.DamageCause.MAGIC,
+            EntityDamageEvent.DamageCause.LIGHTNING
+    );
+
+    private static final boolean globalEnabled;
+
+    private Boolean mapOverride;
```

Comment:
There seems to be an inconsistency with boolean fields. Some use the primitive type and some use the class type